### PR TITLE
feat: Add `erc20_balance_of`

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: src/ethereum_canister_backend/src/lib.rs [hunk 1]
```diff
@@ -66,17 +77,22 @@ async fn setup(
...
-        .map_err(|err| {
-            ic_cdk::println!("Get block number failed: {err}");
-            EthClientError::RpcFailed("get_block_number".into())
-        })?;
+        .map_err(|err| EthClientError::RpcFailed("get_block_number", err.to_string()))?;
 
     Ok(head_block_num.into())
 }
+
+#[update]
```
Comment (zvolin) [inline src/ethereum_canister_backend/src/lib.rs:66]:
```suggestion
#[query(composite = true)]
```
composite allows for inter canister calls in query
https://docs.rs/ic-cdk/latest/ic_cdk/attr.query.html

Comment (oblique) [reply src/ethereum_canister_backend/src/lib.rs:66]: With this change, the `erc20_balance_of` returns this error:

```
Error: Failed query call.
Caused by: Failed query call.
  The replica returned a replica error: Replica Error: reject code CanisterError, reject message IC0504: Canister bkyz2-fmaaa-aaaaa-qaaaq-cai violated contract: "ic0_call_cycles_add128" cannot be executed in non replicated query mode, error code Some("IC0504")
```

File: src/ethereum_canister_backend/src/lib.rs [hunk 2]
```diff
@@ -18,24 +18,14 @@ thread_local! {
     static HELIOS: RefCell<Option<Rc<Client<ConfigDB>>>> = RefCell::new(None);
 }
 
-#[derive(Debug, CandidType, thiserror::Error)]
-enum EthClientError {
-    #[error("Couldn't set up the client")]
-    SetupFailed,
+#[derive(Debug, thiserror::Error)]
+#[error("Rpc method failed: {0}: {1}")]
+struct RpcError(&'static str, String);
```
Comment (zvolin) [inline src/ethereum_canister_backend/src/lib.rs]: it seems unused

Comment (zvolin) [reply src/ethereum_canister_backend/src/lib.rs]: sry, somehow wrongly opened the diff

**Review by zvolin — APPROVED** [review]
LGTM

**Pull request merged**